sphinx/themes/basic/static/searchtools.js

```diff
@@ -345,7 +350,8 @@ const Search = {
     // note the reversing of results, so that in the case of duplicates, the highest-scoring entry is kept
     let seen = new Set();
     results = results.reverse().reduce((acc, result) => {
-      let resultStr = result.slice(0, 4).concat([result[5]]).map(v => String(v)).join(',');
+      const [docName, _title, anchor, descr, _score, filename] = result;
+      const resultStr = `${docName},${anchor},${descr},${filename}`;
```

Comment:
Hmm, I don't see how this would pass the test below. :thinking: The body search and title search will have different `resultStr`'s (since they have different anchors), so you'd see them both in the results.

Reply:
The test does succeed with that - could that imply a bug in the test?

Reply:
There's only one search result because my other fixes haven't been applied (I left them out for ease of review). If you rebase your branch on top of these pull requests the test should start failing:

https://github.com/sphinx-doc/sphinx/pull/11960
https://github.com/sphinx-doc/sphinx/pull/11958

Reply:
Hm.  Are you sure?  I can still get the test to pass with a `docid` of `1` instead of zero, and the partial-matching only affects the scoring, if I understand correctly?

Reply:
Took a look, I think it's actually #11959 (the multiple matches) that allows multiple results to be generated as you'd expect. If you apply this squashed patch your tests should start failing: https://gist.github.com/wlach/74124608c2113d8eb9737cdced4762db

Reply:
Ok, yep - that is subtle.  Thanks.  Re-visiting the approach here with that in place.

Reply:
Why do _two_ `allTitles` entries exist for the same document and section title (for example, `Get started`), but one with a `null` title?

![image](https://github.com/wlach/sphinx/assets/55152140/ed536c0f-4eac-422a-9485-de7353945312)

Reply:
What's the test case here? `alltitles` would be coming from what Sphinx itself generates for the document index. If it's generating duplicate/redundant values I'd say that's a different bug.

Reply:
(note that even if there's a bug there, I'd expect to see duplicate results because both a title and term search can match the same document-- so I don't think that negates the need for this PR)

Reply:
I think there is a bug there on the Python HTML/index build side, yep, and I think it creates duplicate titles.

So there is duplication _within_ `titles`, and then also another duplicate between `titles` and [search term results](https://github.com/wlach/sphinx/blob/5a264654a1ef1e465d82f2603169638b1c71fdfa/sphinx/themes/basic/static/searchtools.js#L339-L340).

A further change may be needed to duplicate between those client-side, yes, but I think it could be simplified (more like e2d07a1008b61017b7568ac813f41c08d6ff1a94) if the duplicate `titles` index entries are removed.

Reply:
Bug raised as sphinx-doc/sphinx#11965.

Reply:
> A further change may be needed to duplicate between those client-side, yes, but I think it could be simplified (more like https://github.com/wlach/sphinx/commit/e2d07a1008b61017b7568ac813f41c08d6ff1a94) if the duplicate titles index entries are removed.

Hmm, I'm confused-- that looks similar to my original solution which you rejected here:

https://github.com/sphinx-doc/sphinx/pull/11942#discussion_r1480599214

Reply:
It's not ideal, but it's smaller -- there's no `isDocumentTitle`, and no additional boolean passed down to the de-duplication phase.

Reply:
(and in fact, it might not even be required.  if the index-construction issue is valid and fixed, then we could try removing it and relying purely on an updated de-duplication key)

Reply:
> It's not ideal, but it's smaller -- there's no isDocumentTitle, and no additional boolean passed down to the de-duplication phase.

Ok, I don't have a super strong opinion: I thought your original critique had a point but I don't think it matters all that much in the end. 

> (and in fact, it might not even be required. if the index-construction issue is valid and fixed, then we could try removing it and relying purely on an updated de-duplication key)

I am pretty sure we still have the problem of the title search / term search returning results. The unit test uses a hand-constructed index without this problem and still reproduces the bug.